### PR TITLE
Exporter/Stats/Stackdriver: Add user agent as client header. 

### PR DESCRIPTION
**Pull request by songy23**
Equivalent to https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/master/stats.go#L80-L81.

See also https://github.com/googleapis/gax-java/blob/a60bd347b69fbf725b0dd8ae18bbcf5b00b66b7b/gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcHeaderInterceptor.java#L63-L67.

### CODE REVIEW

File: exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverStatsExporter.java [hunk 1]
```diff
@@ -317,6 +328,10 @@ private static void createInternal(
         metricServiceClient =
             MetricServiceClient.create(
                 MetricServiceSettings.newBuilder()
+                    .setTransportChannelProvider(
```
Comment (mayurkale22) [inline exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverStatsExporter.java]: can you pls add unit test for this?

Comment (songy23) [reply exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverStatsExporter.java]: SG, done.

**Comment by songy23** [comment]
Thanks for the review. Before I merge this PR I'd like to double check with GAX-Java team that this is the correct way for setting headers. Filed https://github.com/googleapis/gax-java/issues/614 and waiting on their responses.

**Comment by songy23** [comment]
Looks like C++ did this in a similar way to Go: https://github.com/census-instrumentation/opencensus-cpp/pull/234.

**Comment by songy23** [comment]
Quotes from https://github.com/googleapis/gax-java/issues/614#issuecomment-439067683:

> As the comment says ("only merging with gRPC default value for User-Agent is permitted"), the user-agent value is merged with the grpc one.

It seems in Java the custom `user-agent` value works the same way as other languages (merged with the default gRPC value). I've also run a few examples testing the Stackdriver exporter with this change. Everything works fine, though I'm not sure if the `user-agent` value is propagated to Stackdriver backend.

Going to merge this PR first, will change to use `ClientSettings.WithUserAgent()` once GAX-Java team added support for that.

**Comment by g-easy** [comment]
Please let me know if this change was sufficient for Stackdriver and also if the format is correct.

**Comment by songy23** [comment]
> Please let me know if this change was sufficient for Stackdriver and also if the format is correct.

/cc @rakyll I remember you mentioned there's a way to verify the "user-agent" header in Stackdriver backend. Maybe we can chat offline to see if is can be applied to all languages.

**Pull request merged**